tree_test.go

```diff
@@ -1165,6 +1166,80 @@ func TestOverwrite(t *testing.T) {
 	require.NoError(err, "SaveVersion should not fail, overwrite was idempotent")
 }
 
+func TestLoadVersionForOverwriting(t *testing.T) {
+	require := require.New(t)
+
+	mdb := db.NewMemDB()
+	tree := NewMutableTree(mdb, 0)
+
+	maxLength := 100
+	count := 1
+	for {
+		countStr := strconv.Itoa(count)
+		// Set one kv pair and save version
+		tree.Set([]byte("key"+countStr), []byte("value"+countStr))
+		_, _, err := tree.SaveVersion()
+		require.NoError(err, "SaveVersion should not fail")
+		count++
+		if count > maxLength {
+			break
+		}
```

Comment:
nitpicking: use a classical for loop instead of `break` to increase readability. `for count:=1; count <= maxLength; count++ {}`

Reply:
Good suggestion. I will modify the code.